paimon-core/src/main/java/org/apache/paimon/iceberg/manifest/IcebergConversions.java

```diff
@@ -112,6 +131,15 @@ public static Object toPaimonObject(DataType type, byte[] bytes) {
                 DecimalType decimalType = (DecimalType) type;
                 return Decimal.fromUnscaledBytes(
                         bytes, decimalType.getPrecision(), decimalType.getScale());
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                int timestampPrecision = ((TimestampType) type).getPrecision();
+                long timestampLong =
```

Comment:
Change the order of these two statements, for example:

Preconditions.checkArgument(
                        timestampPrecision > 3 && timestampPrecision <= 6,
                        "Paimon Iceberg compatibility only support timestamp type with precision from 4 to 6.");
                long timestampLong =
                        ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN).getLong();